researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/factory/TaskFactory.java

```diff
@@ -0,0 +1,35 @@
+package org.sagebionetworks.bridge.researchstack.factory;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import org.researchstack.skin.model.TaskModel;
+import org.researchstack.skin.task.SmartSurveyTask;
+
+/**
+ * This class encapsulates creating Task instances, to allow us to separate the logic of creating
+ * Tasks from using tasks. This helps simplify tests.
+ */
+public class TaskFactory {
```

Comment:
Did you look at SurveyFactory or TaskItemFactory? I'd suggest just adding a method in SurveyFactory for now with newSmartSurveyTask and using that class.